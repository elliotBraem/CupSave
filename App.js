/* global window */
import React from 'react';
import {Provider} from 'react-redux';
import {PersistGate} from 'redux-persist/integration/react';
import {AppRegistry} from 'react-native';

import {ReactReduxFirebaseProvider} from 'react-redux-firebase';
import {createFirestoreInstance} from 'redux-firestore';

import firebase from 'firebase/app';
import 'firebase/auth';
import 'firebase/firestore';
import 'firebase/storage';

import {firebaseConfig, reactReduxFirebaseConfig} from './src/constants/firebase';
import configureStore from './src/store/configureStore';
import Loading from './src/components/Loading';
import AppContainer from './src/navigation/AppNavigation';

const initialState = window.__INITIAL_STATE__ || {
  firebase: {
    authError: null,
  },
};

const {store, persistor} = configureStore(initialState);

// Initialize Firebase
// Initialize Firebase only if an fbInstance was not passed to the window (tests)
if (!window.fbInstance) {
  firebase.initializeApp(firebaseConfig);
}

// Initialize firestore service on firebase instance
firebase.firestore();

const App = () => (
  <Provider store={store}>
    <ReactReduxFirebaseProvider
      config={reactReduxFirebaseConfig}
      dispatch={store.dispatch}
      firebase={firebase}
      createFirestoreInstance={createFirestoreInstance}>
      <PersistGate loading={<Loading />} persistor={persistor}>
        <AppContainer />
      </PersistGate>
    </ReactReduxFirebaseProvider>
  </Provider>
);

AppRegistry.registerComponent('CupSave', () => App, false);

export default App;
