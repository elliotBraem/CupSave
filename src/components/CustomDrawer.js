/* eslint-disable react/prop-types */
import React, {Component} from 'react';
import {Text, StyleSheet, View, Image, TouchableOpacity} from 'react-native';
import {DrawerItems, SafeAreaView} from 'react-navigation';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';
import Logout from '../assets/images/drawer-icons/logout-icon.svg';
import COLORS from '../constants/colors';
import * as authActions from '../store/actions/auth';

const profileImage = require('../assets/images/profileicon.png');

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  headerContainer: {
    marginLeft: 20,
    marginBottom: 20,
  },
  headerText: {
    fontSize: 14,
    color: COLORS.white,
  },
  profileImage: {
    width: 80,
    height: 80,
    marginBottom: 20,
    borderRadius: 40,
  },
  bottom: {
    justifyContent: 'flex-end',
    flex: 1,
    marginBottom: 36,
    marginLeft: 20,
  },
  innerBottom: {
    flexDirection: 'row',
  },
  logout: {
    marginLeft: 10,
    color: COLORS.white,
    fontSize: 16,
  },
});

class CustomDrawer extends Component {
  static propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
    }).isRequired,
    auth: PropTypes.shape({
      isAuthenticated: PropTypes.bool.isRequired,
    }).isRequired,
    logout: PropTypes.func.isRequired,
  };

  state = {errorMessage: null};

  handleLogout = async () => {
    const {logout, navigation} = this.props;
    await logout()
      .then(navigation.navigate('Auth'))
      .catch(error => {
        console.log(error);
        this.setState({errorMessage: error});
      });
  };

  render() {
    const {logout, auth, ...props} = this.props;
    return (
      <SafeAreaView style={styles.container} forceInset={{top: 'always', horizontal: 'never'}}>
        <View style={styles.headerContainer}>
          <Image source={profileImage} style={styles.profileImage} />
          <Text style={styles.headerText}>{auth.user.email}</Text>
        </View>
        <View>
          <DrawerItems {...props} />
        </View>
        <View style={styles.bottom}>
          <TouchableOpacity onPress={this.handleLogout} style={styles.innerBottom}>
            <Logout style={{width: 20, height: 20}} />
            <Text style={styles.logout}>Log out</Text>
          </TouchableOpacity>
        </View>
      </SafeAreaView>
    );
  }
}

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    logout: () => dispatch(authActions.dbLogout()),
  };
};

const mapStateToProps = (state, ownProps) => {
  const auth = state.auth || {};
  return {
    auth,
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CustomDrawer);
