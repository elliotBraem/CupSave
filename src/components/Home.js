import React from 'react';

import {View, Text, StyleSheet} from 'react-native';

const Home = ({todos, addTodo, onNewTodoChange, newTodoText}) => (
  <View style={styles.container}>
    <Text style={styles.header}>Hello</Text>
  </View>
);

export default Home;

const styles = StyleSheet.create({
  container: {
    marginTop: 100,
    paddingLeft: 15,
    paddingRight: 15,
  },
  header: {
    alignSelf: 'center',
    marginBottom: 50,
  },
});
