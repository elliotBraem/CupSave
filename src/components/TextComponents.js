import React from 'react';
import {Text, StyleSheet} from 'react-native';
import COLORS from '../constants/colors';

const styles = StyleSheet.create({
  normalText: {
    fontFamily: 'open-sans-regular',
    fontSize: 22,
    color: COLORS.white,
  },
  statText: {
    fontSize: 24,
    color: COLORS.secondary,
  },
  statSubtext: {
    fontSize: 12,
  },
  titleText: {
    fontSize: 16,
  },
  headerTitle: {
    fontSize: 20,
    color: COLORS.white,
  },
  worldCounterText: {
    fontFamily: 'open-sans-italic',
    color: COLORS.white,
    fontSize: 18,
    marginTop: 20,
  },
  saveACupText: {
    fontFamily: 'open-sans-bold',
    marginTop: 40,
    color: COLORS.primary,
    fontSize: 48,
    paddingRight: 60,
  },
  saveTheWorldText: {
    fontFamily: 'open-sans-italic',
    color: COLORS.white,
    fontSize: 48,
    paddingLeft: 40,
  },
  aboutUsText: {
    fontFamily: 'open-sans',
    color: COLORS.white,
    fontSize: 20,
    paddingLeft: 10,
  },
  aboutUsAuthorText: {
    fontFamily: 'open-sans-italic',
    color: COLORS.secondary,
    fontSize: 14,
    paddingLeft: 10,
  },
});

export const AppText = props => {
  return <Text style={styles.normalText}>{props.children}</Text>;
};

export const StatText = props => {
  return <Text style={styles.statText}>{props.children}</Text>;
};

export const StatSubtext = props => {
  return <Text style={styles.statSubtext}>{props.children}</Text>;
};

export const TitleText = props => {
  return <Text style={styles.titleText}>{props.children}</Text>;
};

export const HeaderTitle = props => {
  return <Text style={styles.headerTitle}>{props.children}</Text>;
};

export const WorldCounterText = props => {
  return <Text style={styles.worldCounterText}>{props.children}</Text>;
};

export const SaveACupText = props => {
  return <Text style={styles.saveACupText}>{props.children}</Text>;
};

export const SaveTheWorldText = props => {
  return <Text style={styles.saveTheWorldText}>{props.children}</Text>;
};

export const AboutUsText = props => {
  return <Text style={styles.aboutUsText}>{props.children}</Text>;
};

export const AboutUsAuthorText = props => {
  return <Text style={styles.aboutUsAuthorText}>{props.children}</Text>;
};
