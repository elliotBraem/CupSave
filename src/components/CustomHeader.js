import React from 'react';
import {StyleSheet, View, Platform} from 'react-native';
import PropTypes from 'prop-types';
import {compose} from 'recompose';
import {withNavigation} from 'react-navigation';
import BurgerIcon from '../assets/images/drawer-icons/burger-icon.svg';
import {HeaderTitle} from './TextComponents';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
    width: '100%',
    height: Platform.OS === 'ios' ? 70 : 70 - 24,
    marginTop: Platform.OS === 'ios' ? 30 : 30 - 24,
    alignContent: 'center',
    flexDirection: 'row',
  },
  burgerIcon: {
    width: 30,
    height: 30,
    marginLeft: 20,
    marginRight: 20,
  },
});

const CustomHeader = ({navigation, title}) => {
  return (
    <View style={styles.container}>
      <BurgerIcon style={styles.burgerIcon} onPress={() => navigation.toggleDrawer()} />
      <HeaderTitle style={styles.headerTitle}>{title}</HeaderTitle>
    </View>
  );
};

CustomHeader.propTypes = {
  navigation: PropTypes.shape({
    toggleDrawer: PropTypes.func.isRequired,
  }).isRequired,
  title: PropTypes.string.isRequired,
};

const enhance = compose(withNavigation);

export default enhance(CustomHeader);
