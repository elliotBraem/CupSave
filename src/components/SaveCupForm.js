import React from 'react';
import {View, StyleSheet, Image, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';
import {withHandlers, compose} from 'recompose';
import {connect} from 'react-redux';
import {withFirebase, withFirestore} from 'react-redux-firebase';
import COLORS from '../constants/colors';
import {AppText} from './TextComponents';
import Logo from '../assets/images/logo.png';

const styles = StyleSheet.create({
  container: {
    width: 300,
    height: 130,
    borderRadius: 12,
    backgroundColor: COLORS.white,
    alignItems: 'center',
  },
  button: {
    marginTop: 10,
    width: 270,
    height: 70,
    borderRadius: 12,
    backgroundColor: COLORS.primary,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  logoIcon: {
    height: 45,
    width: 30,
    marginRight: 30,
  },
});

const SaveCupForm = ({onSaveCupFormSubmit}) => (
  <View style={styles.container}>
    <TouchableOpacity onPress={onSaveCupFormSubmit} style={styles.button}>
      <Image source={Logo} style={styles.logoIcon} />
      <AppText>SAVE A CUP</AppText>
    </TouchableOpacity>
  </View>
);

SaveCupForm.propTypes = {
  onSaveCupFormSubmit: PropTypes.func.isRequired,
};

const enhance = compose(
  withFirebase,
  withFirestore,
  connect(({firebase: {auth}}) => ({
    auth,
  })),
  withHandlers({
    onSaveCupFormSubmit: props => () => {
      const currentUID = props.auth.uid;
      const userRef = props.firestore.collection('users').doc(`${currentUID}`);
      props.firestore
        .runTransaction(async transaction => {
          const doc = await transaction.get(userRef);

          const currentTimeInUnixEpoch = new Date().valueOf();

          const newTotal = doc.exists ? doc.data().consumption.total + 1 : 1;

          const data = {
            consumption: {
              total: newTotal,
              most_recent_consumption: currentTimeInUnixEpoch,
              history: {
                [currentTimeInUnixEpoch]: newTotal,
              },
            },
          };

          transaction.update(userRef, data);
        })
        .catch(error => {
          this.setState({errorMessage: error.message});
        });
    },
  })
);

export default enhance(SaveCupForm);
