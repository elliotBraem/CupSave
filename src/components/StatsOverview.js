import React from 'react';
import {View, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';
import COLORS from '../constants/colors';
import {StatText} from './TextComponents';

const styles = StyleSheet.create({
  container: {
    width: 350,
    height: 90,
    backgroundColor: COLORS.white,
    borderRadius: 12,
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    alignItems: 'center',
  },
});

const StatsOverview = ({drinkSize, totalCupsSaved, level}) => (
  <View style={styles.container}>
    <StatText style={styles.statText}>{drinkSize}</StatText>
    <StatText style={styles.statText}>{totalCupsSaved}</StatText>
    <StatText style={styles.statText}>{level}</StatText>
  </View>
);

StatsOverview.propTypes = {
  drinkSize: PropTypes.number,
  totalCupsSaved: PropTypes.number,
  level: PropTypes.number,
};

StatsOverview.defaultProps = {
  drinkSize: 16,
  totalCupsSaved: 0,
  level: 0,
};

export default StatsOverview;
