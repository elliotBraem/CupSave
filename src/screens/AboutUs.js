import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {StyleSheet, View, Platform} from 'react-native';
import CustomHeader from '../components/CustomHeader';
import COLORS from '../constants/colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between',
    backgroundColor: COLORS.primary,
  },
  inner: {
    marginTop: Platform.OS === 'ios' ? 100 : 100 - 24,
  },
});

class AboutUsScreen extends PureComponent {
  static propTypes = {
    navigation: PropTypes.shape({
      openDrawer: PropTypes.func.isRequired,
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  render() {
    return (
      <View style={styles.container}>
        <CustomHeader title="About Us" />
        <View style={styles.header} />
      </View>
    );
  }
}

export default AboutUsScreen;
