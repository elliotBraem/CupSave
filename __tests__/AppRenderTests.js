import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import CustomDrawer from '../src/components/CustomDrawer';
import CustomHeader from '../src/components/CustomHeader';
import LoadingComp from '../src/components/Loading';
import SaveCupForm from '../src/components/SaveCupForm';
import * as TextComponents from '../src/components/TextComponents';
import StatsOverview from '../src/components/StatsOverview';
import AboutUs from '../src/screens/AboutUs';
import Home from '../src/screens/Home';
import LoadingScreen from '../src/screens/Loading';
import Login from '../src/screens/Login';
import Map from '../src/screens/Map';
import Profile from '../src/screens/Profile';
import QRScanner from '../src/screens/QRScanner';
import Settings from '../src/screens/Settings';
import SignUp from '../src/screens/SignUp';
import Stats from '../src/screens/Stats';

// App
it('App renders without crashing', () => {
  const rendered = renderer.create(<App />).toJSON();
  expect(rendered).toBeTruthy();
});

it('App test against snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> CustomDrawer
it('CustomDrawer renders without crashing', () => {
  const rendered = renderer.create(<CustomDrawer />).toJSON();
  expect(rendered).toBeTruthy();
});

it('CustomDrawer test against snapshot', () => {
  const tree = renderer.create(<CustomDrawer />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> CustomHeader
it('CustomHeader renders without crashing', () => {
  const rendered = renderer.create(<CustomHeader />).toJSON();
  expect(rendered).toBeTruthy();
});

it('CustomHeader test against snapshot', () => {
  const tree = renderer.create(<CustomHeader />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> Loading
it('Loading renders without crashing', () => {
  const rendered = renderer.create(<LoadingComp />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Loading test against snapshot', () => {
  const tree = renderer.create(<LoadingComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> SaveCupForm
it('SaveCupForm renders without crashing', () => {
  const rendered = renderer.create(<SaveCupForm />).toJSON();
  expect(rendered).toBeTruthy();
});

it('SaveCupForm test against snapshot', () => {
  const tree = renderer.create(<SaveCupForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> StatsOverview
it('StatsOverview renders without crashing', () => {
  const rendered = renderer.create(<StatsOverview />).toJSON();
  expect(rendered).toBeTruthy();
});

it('StatsOverview test against snapshot', () => {
  const tree = renderer.create(<StatsOverview />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Components -> TextComponents
it('TextComponents renders without crashing', () => {
  const rendered = renderer.create(<TextComponents />).toJSON();
  expect(rendered).toBeTruthy();
});

it('TextComponents test against snapshot', () => {
  const tree = renderer.create(<TextComponents />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> AboutUs
it('AboutUs renders without crashing', () => {
  const rendered = renderer.create(<AboutUs />).toJSON();
  expect(rendered).toBeTruthy();
});

it('AboutUs test against snapshot', () => {
  const tree = renderer.create(<AboutUs />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Home
it('Home renders without crashing', () => {
  const rendered = renderer.create(<Home />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Home test against snapshot', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Loading
it('Loading renders without crashing', () => {
  const rendered = renderer.create(<LoadingScreen />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Loading test against snapshot', () => {
  const tree = renderer.create(<LoadingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Login
it('Login renders without crashing', () => {
  const rendered = renderer.create(<Login />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Login test against snapshot', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Map
it('Map renders without crashing', () => {
  const rendered = renderer.create(<Map />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Map test against snapshot', () => {
  const tree = renderer.create(<Map />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Profile
it('Profile renders without crashing', () => {
  const rendered = renderer.create(<Profile />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Profile test against snapshot', () => {
  const tree = renderer.create(<Profile />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> QRScanner
it('QRScanner renders without crashing', () => {
  const rendered = renderer.create(<QRScanner />).toJSON();
  expect(rendered).toBeTruthy();
});

it('QRScanner test against snapshot', () => {
  const tree = renderer.create(<QRScanner />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Settings
it('Settings renders without crashing', () => {
  const rendered = renderer.create(<Settings />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Settings test against snapshot', () => {
  const tree = renderer.create(<Settings />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> SignUp
it('SignUp renders without crashing', () => {
  const rendered = renderer.create(<SignUp />).toJSON();
  expect(rendered).toBeTruthy();
});

it('SignUp test against snapshot', () => {
  const tree = renderer.create(<SignUp />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Screens -> Stats
it('Stats renders without crashing', () => {
  const rendered = renderer.create(<Stats />).toJSON();
  expect(rendered).toBeTruthy();
});

it('Stats test against snapshot', () => {
  const tree = renderer.create(<Stats />).toJSON();
  expect(tree).toMatchSnapshot();
});
