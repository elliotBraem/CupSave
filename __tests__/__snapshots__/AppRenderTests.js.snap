// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#79db85"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Loading test against snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#79db85"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`SignUp test against snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 50,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 38,
          "marginBottom": 20,
        }
      }
    >
      Welcome to CupSave!
    </Text>
    <Text>
      Sign Up
    </Text>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "alignSelf": "stretch",
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "marginTop": 14,
          "textAlign": "center",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "stretch",
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "marginTop": 14,
          "textAlign": "center",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="Confirm Password"
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "stretch",
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 40,
          "marginTop": 14,
          "textAlign": "center",
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 10,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Already have an account? Login
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`StatsOverview test against snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 12,
      "flexDirection": "row",
      "height": 90,
      "justifyContent": "space-evenly",
      "width": 350,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1A1A1A",
        "fontSize": 24,
      }
    }
  >
    16oz
  </Text>
  <Text
    style={
      Object {
        "color": "#1A1A1A",
        "fontSize": 24,
      }
    }
  >
    0
  </Text>
  <Text
    style={
      Object {
        "color": "#1A1A1A",
        "fontSize": 24,
      }
    }
  >
    0
  </Text>
</View>
`;
